### PR TITLE
deprecated @Module properties

### PR DESCRIPTION
**Pull request by fourlastor**
Hi,

I saw this was still in to-do for the release so I thought I might as well help.
I removed the documentation also to make it more clear that it shouldn't be used.

### CODE REVIEW

**Comment by fourlastor** [comment]
Just signed the contributor agreement as an independent developer.

File: core/src/main/java/dagger/Module.java [hunk 1]
```diff
@@ -26,73 +26,24 @@
...
+  @Deprecated
   boolean overrides() default false;
 
-  /**
-   * Additional {@code @Module}-annotated classes from which this module is
-   * composed. The de-duplicated contributions of the modules in
-   * {@code includes}, and of their inclusions recursively, are all contributed
-   * to the object graph.
-   */
+  @Deprecated
```
Comment (JakeWharton) [inline core/src/main/java/dagger/Module.java]: Revert. This is still used and useful.

**Comment by fourlastor** [comment]
My bad, fixed.

**Comment by cgruber** [comment]
We may need to do this internally, as we're not turning on deprecation on dagger1 internally, since it slows down the builds of really large projects (seriously - the extra i/o is a problem). So we have to do some fancy magic with MOE to do things differently between the two repos.  Let me consult a bit though.  We might yet use this pull.

**Comment by cgruber** [comment]
It looks like we're going to actually delete them, but doing so via an internal commit, for all sorts of practical syncing reasons. Sorry, though thank you very much for this pull.

**Pull request closed without merging**